lib/rust/ensogl/core/src/display/shape/primitive/system/cached.rs

```diff
@@ -0,0 +1,488 @@
+//! Shape Systems with instance cached on global texture
+//!
+//! These systems are a valid [shape systems](super) which cannot be parameterized, but have
+//! a single instance cached on a special texture (available in Glsl as `pass_cached_shapes`). The
+//! texture keeps the color (except alpha) and the signed distance. The cached instance may be later
+//! used inside other shape systems by using [`AnyCachedShape`].
+//!
+//! # Limitations
+//!
+//! * The cached shapes does not support alpha channels fully: during rendering to the texture the
+//!   alpha value is dropped. See _Cached Shapes Texture_ section for details.
+//! * The signed distance kept in the texture is capped at [`CACHED_TEXTURE_MAX_DISTANCE`]. This may
+//!   give aliasing effects when zoomed out more than this distance.
+//! * The shapes rendered from the texture may not always gives the best quality results, see
+//!   _Limitations_ section of [`AnyCachedShape`] docs.
+//!
+//! # Usages
+//!
+//! The main idea behind cached shapes is speeding up rendering applications with many shape
+//! systems. Because each shape system is drawn with a different draw call, having e.g. a grid of
+//! icons can increase number od those, greatly worsening the application performance. But instead
+//! all the icons can be rendered to texture, and the icon grid can use only one shape system
+//! parameterized by the exact icon id.
```

Comment:
Some minor English issues here and in other places - could you put it in ChatGPT and ask it to rewrite it nicely, please?